src/container.ts

```diff
@@ -26,54 +29,114 @@ export type ContainerEventArgs = EventArgs | WindowEventArgs | LayoutEventArgs;
  * @extends ContainerWindowManager
  * @extends ContainerNotificationManager
  */
-export interface Container extends ContainerWindowManager, ContainerNotificationManager {
+export abstract class Container extends EventEmitter implements ContainerWindowManager, ContainerNotificationManager {
+    private static _ipc: MessageBus; // tslint:disable-line
+
     /**
      * Display type of current Container.
      * @type {string}
      */
-    hostType: string;
+    public abstract hostType: string;
 
     /**
      * Unique v4 GUID for this Container instance
      * @type {string}
      */
-    uuid: string;
+    public abstract uuid: string;
+
+    private static staticEmitter: EventEmitter = new EventEmitter();
+
+    public abstract getMainWindow(): ContainerWindow;
+
+    public abstract getCurrentWindow(): ContainerWindow;
+
+    public abstract createWindow(url: string, options?: any): ContainerWindow;
+
+    public abstract saveLayout(name: string): Promise<PersistedWindowLayout>;
+
+    public abstract loadLayout(name: string): Promise<PersistedWindowLayout>;
+
+    public abstract getLayouts(): Promise<PersistedWindowLayout[]>;
 
     /**
      * Adds an icon and context menu to the system notification area.
      * @param {TrayIconDetails} details Details for the tray icon.
      * @param listener (Optional) Callback for when the tray icon is clicked.
      * @param {MenuItem[]} menuItems (Optional) Context menu.
      */
-    addTrayIcon(details: TrayIconDetails, listener?: () => void, menuItems?: MenuItem[]);
+    public abstract addTrayIcon(details: TrayIconDetails, listener?: () => void, menuItems?: MenuItem[]);
+
+    public abstract showNotification(title: string, options?: NotificationOptions);
+
+    public abstract getAllWindows(): Promise<ContainerWindow[]>;
```

Comment:
This might look weird at first but it is defining ipc property as static and instance. The instance one decorates the static.